erpnext/hr/doctype/leave_ledger_entry/leave_ledger_entry.py

```diff
@@ -183,4 +183,64 @@ def expire_carried_forward_allocation(allocation):
 			from_date=allocation.to_date,
 			to_date=allocation.to_date
 		)
-		create_leave_ledger_entry(allocation, args)
\ No newline at end of file
+		create_leave_ledger_entry(allocation, args)
+
+
+def get_leave_allocation(employee, leave_type, date, fields=None):
+	allocation = frappe.db.get_value('Leave Allocation',
+		filters={
+			'employee': employee,
+			'leave_type': leave_type,
+			'from_date': ['<=', date],
+			'to_date': ['>=', date],
+			'docstatus': 1
+		},
+		fieldname=fields or "name",
+		as_dict=True if isinstance(fields, list) else False
+	)
+	if not allocation:
+		frappe.throw(_("Leave Allocation not found."))
```

Comment:
The location of this `frappe.throw`. Can it `return None` if not found and let the caller throw the error when required instead?